container/container.go

```diff
@@ -42,6 +43,9 @@ type SubcontainerEvent struct {
 
 	// The full container name of the container where the event occurred.
 	Name string
+
+	//who detected it
+	DetectionType string
```

Comment:
I'm trying to minimize the code duplication, so only have them all going through the same event loop which goes to the same factory loop.

However, since we are dealing with the same container names if we don't distinguish where the container name came from, rkt wouldn't be able to determine if it was for it or not and would have to wait to make sure.  it's just labeling which watcher found it.

I'm not convinced this is right, but I'm trying to keep the code simple and not overly complicated